### PR TITLE
Change mechanism to retrieve the certificate from assembly

### PR DESCRIPTION
**Pull request by saintentropy**
### Purpose

This PR update the mechanism the CertificateVerification code uses to extract a certificate from an assembly during the verification process

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner @QilongTang 

### FYIs

@sm6srw

### CODE REVIEW

**Comment by QilongTang** [comment]
@jnealb  Please work with @smangarole on the testing.

**Comment by QilongTang** [comment]
@saintentropy Please test this with the master build today, then cherry-pick to RC2.5.0_master branch

**Comment by mjkkirschner** [comment]
@saintentropy - it seems that the tests you added to core need to be added to DynamoRevit.
also - 
-[ ] please verify this blocks unsigned dlls or invalid certs.

File: src/DynamoUtilities/CertificateVerification.cs [hunk 1]
```diff
@@ -37,7 +38,17 @@ public static bool CheckAssemblyForValidCertificate(string assemblyPath)
             }
 
             //Verify the node library has a verified signed certificate
-            var cert = asm.Modules.FirstOrDefault()?.GetSignerCertificate();
+            X509Certificate cert;
```
Comment (tamasbadics) [inline src/DynamoUtilities/CertificateVerification.cs]: I think it may be even safer to check this before loading the asm on line 32. (Just a comment)

Comment (QilongTang) [reply src/DynamoUtilities/CertificateVerification.cs]: FYI: @saintentropy What do you think? Would you consider such change in `master` branch?

**Pull request merged**